### PR TITLE
PT-415: Google Form for announcements

### PR DESCRIPTION
**Pull request by florianmski**
Tracked in JIRA by [PT-415](https://novoda.atlassian.net/browse/PT-415)

### Scope of the PR

In order to be able to use Google Form to feed the spreadsheet, we needed to expose two new parameters for the `announcement.js` widget, namely:
- `sheetName` which is the name of the sheet we use to gather the form response (no to mistake with the actual spreadsheet file, a spreadsheet file can contains multiple sheets)
- `range` which is where the data is situated in the spreadsheet

As a side note, for the Google form we decided not to use the default "date and time" widget as it uses the date formatting corresponding to the user local. We also tried for some time to fix it on the spreadsheet side but it doesn't do it dynamically. As in, when we format the existing cells it works but when new one are added the formatting doesn't take place. All in all instead of using some script or hacks trickery and to keep it as simple as possible the Google form explain how to enter the date in the expected format which in the end is really not that hard :)

Paired with @tasomaniac on the spreadsheet format craziness

### CODE REVIEW

**Comment by florianmski** [comment]
@tasomaniac I added a comment in the `sample.json` 👍

File: node-server/server/configs/sample.json [hunk 1]
```diff
@@ -49,7 +49,14 @@
     "venueId": "56f0fe62cd107047e6c74bac"
   },
   "announcement": {
-    "spreadsheetId": "YOUR_GOOGLE_SPREADSHEET_ID"
+    "spreadsheetId": "YOUR_GOOGLE_SPREADSHEET_ID",
+    "sheetName": "SHEET_USED_IN_THE_SPREADSHEET",
+    "range": "XX:XX",
+    "__comment": "For the range you can use for instance A2:F to capture all rows from column A to column F starting at row 2.
```
Comment (frapontillo) [inline node-server/server/configs/sample.json]: 👍

File: node-server/server/configs/sample.json [hunk 2]
```diff
@@ -49,7 +49,14 @@
...
+    The order of the columns needs to be title, content, startDate, endDate with startDate and endDate being in the ISO format.
```
Comment (frapontillo) [inline node-server/server/configs/sample.json]: But you can't have new lines in JSON strings. You can either use `\n` or simulate a new line by setting `__comment` as an array.

File: node-server/server/configs/sample.json (hunk 1 above)
Comment (eduardb) [inline node-server/server/configs/sample.json]: Shouldn't the example be with `C2:F`? As in, there are 4 columns that need to be captured.

Comment (florianmski) [inline node-server/server/configs/sample.json]: 👀

**Comment by tasomaniac** [comment]
@eduardb your turn to merge this bad boi. 
Then no one can stop us from submitting crazy announcements via form.

**Pull request merged**